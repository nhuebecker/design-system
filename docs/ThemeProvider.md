# ThemeProvider

The ThemeProvider component is a wrapper around the styled-componentsâ€™ [ThemeProvider][1]

Wrap the root of your application with the `ThemeProvider` component,
which adds the Design System theme to context for use in styled-components
and sets typographic defaults (including, optionally, loading our webfont, Averta).
This should only be included once in your application.

```jsx
<ThemeProvider webfonts>
  <Heading>Hello</Heading>
</ThemeProvider>
```

[1]:	https://www.styled-components.com/docs/advanced#theming
