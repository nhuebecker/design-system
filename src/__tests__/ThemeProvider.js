import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider, theme } from '..'
import { Base } from '../ThemeProvider'
import styled from 'styled-components'

const Content = styled.p`
  color: ${({ theme }) => theme.colors.primary};
`

describe('ThemeProvider', () => {
  test('renders content with theme', () => {
    const json = renderer.create(
      <ThemeProvider>
        <Content>Hello!</Content>
      </ThemeProvider>
    )
    expect(json).toMatchSnapshot()
  })
})
