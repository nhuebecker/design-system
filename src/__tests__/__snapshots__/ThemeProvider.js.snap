// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeProvider renders content with theme 1`] = `
.c0 {
  color: #e42d42;
}

<p
  className="c0"
>
  Hello!
</p>
`;
